app/WebParsing/ReqParser.hs

```diff
@@ -26,6 +26,32 @@ completionPrefix = Parsec.choice (map (Parsec.try . Parsec.string) [
     ])
     >> Parsec.spaces
 
+programPrefix :: Parser ()
+programPrefix = Parsec.choice (map caseInsensitiveStr [
+    "admission to",
+    "enrolment in the ",
+    "enrolment in an ",
+    "enrolment in a ", -- trailing space to prevent matching "enrolment in A..."
+    "enrolment in"
+    ])
+    >> Parsec.spaces
+
+degreeType :: Parser String
+degreeType = do
+    Parsec.spaces
+    degree <- Parsec.choice $ map caseInsensitiveStr [
+        "major",
+        "specialist"
+        ]
+    Parsec.spaces
+    return degree
+
+programSuffix :: Parser String
+programSuffix = Parsec.choice $ map caseInsensitiveStr [
+    " program of study",
```

Comment:
See comment above about spaces. Try to find a different way of checking for spaces other than inserting it into the string literal.